core/src/main/java/com/linecorp/armeria/server/file/NonExistentHttpFile.java

```diff
@@ -29,16 +31,18 @@
 
 final class NonExistentHttpFile implements HttpFile {
 
-    static final NonExistentHttpFile INSTANCE = new NonExistentHttpFile(null);
+    static final NonExistentHttpFile INSTANCE = new NonExistentHttpFile(null, false);
```

Comment:
Should we migrate all internal usage of `HttpFile.nonExistent()` into `HttpFile.nonExistent(filename)` if possible?

Reply:
As I understand it, that means removing `HttpFile.nonExistent()` and modifying the classes that use it, right?

Reply:
Hello, @ikhoon sorry for the hassle ☺️
If you have a moment, I'd love some hints to solve the problem!

Reply:
https://github.com/line/armeria/blob/f2116739018c15f230dd8f0e5a4dd5db1fefc888/core/src/main/java/com/linecorp/armeria/server/file/CachingHttpFile.java#L93-L93
https://github.com/line/armeria/blob/f2116739018c15f230dd8f0e5a4dd5db1fefc888/core/src/main/java/com/linecorp/armeria/server/file/HttpFile.java#L189
What I meant was these^ two points could be migrated to take advantage of the new API.

Reply:
Being able to provide hints makes life easier for users! Let's see what we need to fix. ☺️